src/support/z_context.erl

```diff
@@ -428,6 +429,41 @@ abs_url(Url, Context) ->
     has_url_protocol(_) ->
         false.
 
+%% @doc Ensure that the URL is for this host and that is sanitized.
+-spec site_url(undefined|string()|binary(), z:context()) -> binary().
+site_url(None, Context) when
+    None =:= undefined;
+    None =:= "";
+    None =:= <<>> ->
+    z_context:abs_url(<<"/">>, Context);
+site_url("#" ++ _ = Frag, _Context) ->
+    z_convert:to_binary(Frag);
+site_url(<<"#", _/binary>> = Frag, _Context) ->
+    Frag;
+site_url("/" ++ _ = Path, Context) ->
+    abs_url(Path, Context);
+site_url(<<"/", _/binary>> = Path, Context) ->
+    abs_url(Path, Context);
+site_url(Url, Context) ->
+    case z_convert:to_list(z_html:noscript(Url)) of
+        "#" ++ _ = Url1 ->
+            z_convert:to_binary(Url1);
+        "/" ++ _ = Url1 ->
+            abs_url(Url1, Context);
+        Url1 ->
+            Path1 = case mochiweb_util:urlsplit(Url1) of
+                {_Scheme, _Netloc, Path, "", ""} ->
+                    Path;
+                {_Scheme, _Netloc, Path, Query, ""} ->
+                    Path ++ "?" ++ Query;
+                {_Scheme, _Netloc, Path, "", Frag} ->
+                    Path ++ "#" ++ Frag;
+                {_Scheme, _Netloc, Path, Query, Frag} ->
+                    Path ++ "?" ++ Query ++ "#" ++ Frag
+            end,
+            abs_url(Path1, Context)
+    end.
+
```

Comment:
Script URLs need to start with something like `javascript:`  (and variations on that because browsers tend to be lax about the syntax here).  So if we add the domain name in front `https://example.com/` then all should be ok (after the sanitize, you have to catch newlines in the URL, which otherwise can add extra headers to the HTTP response).